Update Chef-Sentry cookbook to support Sentry 8.6.0

mattlock12 opened the pull request:
- templates
  add sentry.conf.8.0.py.erb template with structure accomodating to properties
  present in the generated sentry.conf.py file in 8.6
  add conf.yml file to conform to 8.x Sentry standards of splitting configs
  out from sentry.conf.py to conf.yml
- attributes
  update attributes/default.rb to contain default values for new properties in
  sentry.conf.py and conf.yml
- recipes
  update _configure.rb to check for Sentry version and use new templates if
  version > 8

Commit:
Update Chef-Sentry cookbook to support Sentry 8.6.0

- templates
add sentry.conf.8.0.py.erb template with structure accomodating to properties
present in the generated sentry.conf.py file in 8.6
add conf.yml file to conform to 8.x Sentry standards of splitting configs
out from sentry.conf.py to conf.yml

- attributes
update attributes/default.rb to contain default values for new properties in
sentry.conf.py and conf.yml

- recipes
update _configure.rb to check for Sentry version and use new templates if
version > 8

Commit:
update template dirs in attributes and _configure.rb
update _configure recipe to create sentry user per docs

Commit:
update config.yml template to have attr keys for redis config
update _configure to expect data_bag use
user is created with email and password
add shellescape
update recipes/_service to more accurately reflect 8.x
create sv-sentry-[web|worker|cron]-run.erb templates, with log templates
change _service to use config dir for 8.x fork
use explicit guard statement piping in some Django commands to sentry

dhui commented on attributes/default.rb:
```diff
@@ -28,23 +28,38 @@
+# dependencies per: https://docs.getsentry.com/on-premise/server/installation/python/
 default["sentry"]["dependency"]["packages"] = [
+  "python-setuptools",
+  "python-pip",
```
You probably don't `python-pip` or `python-setuptools`.
The [python](https://github.com/poise/python) cookbook should take care of that for you.

dhui commented on attributes/default.rb:
```diff
@@ -57,18 +72,31 @@
+
+# smtp
+default["sentry"]["config"]["smtp_host"] = '0.0.0.0'
+default["sentry"]["config"]["smtp_port"] = '1025'
```
Use 5587 since it's closer to the default of 587

dhui commented on recipes/_configure.rb:
```diff
@@ -58,8 +59,8 @@
 
-(sentry_config["additional_env_vars"] || {}).each do |key, value|
-
+# sent env variables for runit to use
```
sent => set

dhui commented on recipes/_configure.rb:
```diff
@@ -58,8 +59,8 @@
-(sentry_config["additional_env_vars"] || {}).each do |key, value|
-
+# sent env variables for runit to use
+(sentry_config["additional_env_vars"] || {:SENTRY_CONF => node['sentry']['config_dir']}).each do |key, value|
```
Why not update the `additional_env_vars` appropriately?

dhui commented on recipes/_configure.rb:
```diff
@@ -78,93 +79,200 @@
+    subscribes :run, "template[#{initial_admin_json}]", :immediately
+  end
+
+# 8.0 or above
```
Comment should go inside the `else`

dhui commented on templates/default/sentry.conf.8.0.py.erb:
```diff
@@ -0,0 +1,205 @@
+# This file is just Python, with a touch of Django which means
```
How different is this file from `sentry.conf.py.erb`?
Additive config keys should be fine.
And if any config keys have different values between v7 and v8, you can make the appropriate changes before rendering the template.

dhui replied:
you can run `diff -u sentry.conf.py.erb sentry.conf.8.0.py.erb` (or use colordiff on your mac)

mattlock12 replied:
There are quite a few moved/unnecessary values held over from < 8.0, but the main difference is between this template the default template that the 'sentry init' command generates through a manual install. A lot of stuff is moved and/or renamed---so much so that it was confusing for me when I was trying to do the upgrade.
Separating the file remedies the discrepancy with sentry's preferred config layout and (I think) makes it way easier for users to quickly tell which template they should use (and how) for their version

dhui commented on templates/default/sv-sentry-cron-log-run.erb:
```diff
@@ -0,0 +1,2 @@
+#!/bin/sh
```
Why do these sv\* files show up in the diff?
Anyway you can update the existing services to change the command or what's run based on the sentry version?

mattlock12 replied:
I added new sv- templates for the new sentry commands (run|web|cron vs start|celery).
Similarly to the logic above, I wanted to make a clear distinction between versions, rather than patch existing templates. Also similar to how each recipe has a clear if/else block corresponding to < 8.0 & >= 8.0

dhui replied:
In that case, let's create a separate template directory for sentry v8 templates

dhui commented on recipes/_configure.rb:
```diff
@@ -58,8 +59,8 @@
 end
 
+# set env variables for runit to use
 (sentry_config["additional_env_vars"] || {}).each do |key, value|
```
You shouldn't need the `|| {}` if the default for `additional_env_vars` is a hash/mash

mattlock12 replied:
yeah but if somebody uses a data bag and doesn't set it, we don't want errors

dhui replied:
Oh my bad. I thought this was an attribute.

Commit:
split templates out to their own 8.0 directory
update template sources to be more explicit in _service
update conf.yml with system.url-prefix
default to using SMTP settings in sentry.conf.py
update smtp_host to 'reply.getsentry.com'

dhui commented:
:shipit:

Commit:
adjust timeout on sentry-web service to 3 minutes
graceful timeout for uwsgi on worker processes defaults to 60 seconds,
so give sentry enough time to brutally exit worker processes without
causing chef to error

Commit:
update config files to remove 8.x deprecation warnings

Commit:
configure to use google sso
add google sso variables to _configure and sentry.conf.py templates
add google_client_id/secret defaults

dhui commented on recipes/_configure.rb:
```diff
@@ -78,93 +79,203 @@
+      filestore: node["sentry"]["config"]["filestore"],
+      filestore_options: node["sentry"]["config"]["filestore_options"],
+      # google sso
+      google_client_id: node['sentry']['google_client_id'],
```
Is this an attribute or a data bag item?
It looks like an attribute. If so, I'd declare this in the attributes file with a default falsey value.

mattlock12 replied:
That's an attribute set from a data bag
